tests/command_line/test_ssx_integrate.py

```diff
@@ -1,81 +1,90 @@
-from pathlib import Path
-
+import procrunner
 import pytest
 
 from dxtbx.serialize import load
 
 from dials.array_family import flex
-from dials.command_line.ssx_integrate import run as run_integrate
+from dials.command_line.ssx_integrate import run_integration, working_phil
+from dials.util.options import ArgumentParser
 
 # Note that tests are grouped and run serially, to stop many processes trying to
 # extract data from images at same time, which appears to lead to race
 # conditions in some CI jobs.
 
 
 @pytest.mark.xdist_group(name="group1")
-def test_ssx_integrate_stills(dials_data, run_in_tmpdir):
+def test_ssx_integrate_fullprocess(dials_data, tmp_path):
+    # Download data set and the internally referenced images
     ssx = dials_data("cunir_serial_processed", pathlib=True)
-    _ = dials_data("cunir_serial", pathlib=True)
-    indexed_refl = str(ssx / "indexed.refl")
-    indexed_expts = str(ssx / "indexed.expt")
-
-    run_integrate(
+    dials_data("cunir_serial", pathlib=True)
+    result = procrunner.run(
         [
-            indexed_refl,
-            indexed_expts,
-            "algorithm=stills",
+            "dev.dials.ssx_integrate",
+            str(ssx / "indexed.refl"),
+            str(ssx / "indexed.expt"),
             "nproc=1",
-            "json=data.json",
-            "image_range=1:2",
-        ]
+            "batch_size=3",
+            "output.json=data.json",
+            "algorithm=stills",
+        ],
+        working_directory=tmp_path,
     )
+    assert not result.returncode and not result.stderr
+    assert (tmp_path / "integrated_1.refl").is_file()
+    assert (tmp_path / "integrated_1.expt").is_file()
+    assert (tmp_path / "integrated_2.refl").is_file()
+    assert (tmp_path / "integrated_2.expt").is_file()
+    assert (tmp_path / "dials.ssx_integrate.html").is_file()
+    assert (tmp_path / "data.json").is_file()
```

Comment:
```suggestion
    assert tmp_path.joinpath("integrated_1.refl").is_file()
    assert tmp_path.joinpath("integrated_1.expt").is_file()
    assert tmp_path.joinpath("integrated_2.refl").is_file()
    assert tmp_path.joinpath("integrated_2.expt").is_file()
    assert tmp_path.joinpath("dials.ssx_integrate.html").is_file()
    assert tmp_path.joinpath("data.json").is_file()
```
I find `x.y().z()` easier to read than `(x / y).z()`